File: benchmarks/real_comparison/fib.js
// Fibonacci benchmark - Node.js
function fib(n) {
  if (n <= 1) return n;
  return fib(n - 1) + fib(n - 2);
}

// Warmup V8
for (let i = 0; i < 3; i++) {
  fib(35);
}

const start = performance.now();

let result = 0;
for (let i = 0; i < 5; i++) {
  result = fib(40);
}

const end = performance.now();
console.log(
  `Node.js: fib(40) = ${result}, Time: ${(end - start).toFixed(2)} ms (5 runs)`
);

File: benchmarks/real_comparison/fib.rs
// Fibonacci benchmark - Rust
use std::time::Instant;

fn fib(n: i32) -> i64 {
    if n <= 1 {
        return n as i64;
    }
    fib(n - 1) + fib(n - 2)
}

fn main() {
    let start = Instant::now();

    let mut result: i64 = 0;
    for _ in 0..5 {
        result = fib(40);
    }

    let elapsed = start.elapsed();
    println!(
        "Rust: fib(40) = {}, Time: {:.2} ms (5 runs)",
        result,
        elapsed.as_secs_f64() * 1000.0
    );
}
